StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1201ElementsMustAppearInTheCorrectOrder.cs

```diff
@@ -226,6 +248,46 @@ private static void HandleMemberList(SyntaxNodeAnalysisContext context, SyntaxLi
                     continue;
                 }
 
+                bool compareKind = true;
+                for (int j = 0; compareKind && j < kindIndex; j++)
+                {
+                    switch (elementOrder[j])
+                    {
+                    case OrderingTrait.Accessibility:
+                        if (MemberOrderHelper.GetAccessLevelForOrdering(members[i + 1], members[i + 1].GetModifiers())
+                            != MemberOrderHelper.GetAccessLevelForOrdering(members[i], members[i].GetModifiers()))
+                        {
+                            compareKind = false;
+                        }
+
+                        continue;
+
+                    case OrderingTrait.Constant:
+                    case OrderingTrait.Readonly:
+                        // Only fields may be marked const or readonly, and all fields have the same kind.
+                        continue;
+
+                    case OrderingTrait.Static:
+                        bool currentIsStatic = members[i].GetModifiers().Any(SyntaxKind.StaticKeyword);
+                        bool nextIsStatic = members[i + 1].GetModifiers().Any(SyntaxKind.StaticKeyword);
+                        if (currentIsStatic != nextIsStatic)
+                        {
+                            compareKind = false;
+                        }
+
+                        continue;
+
+                    case OrderingTrait.Kind:
+                    default:
+                        continue;
+                    }
+                }
+
+                if (!compareKind)
+                {
+                    continue;
+                }
+
```

Comment:
:thought_balloon: It would be nice is this could be refactored to use the same  previous / current pattern as SA1202/SA1203/SA1204.